apps/native-component-list/src/screens/Contacts/ContactsScreen.tsx

```diff
@@ -83,66 +71,43 @@ export default class ContactsScreen extends React.Component<NavigationScreenProp
       pageOffset,
     });
 
-    const { data: contacts, hasPreviousPage, hasNextPage } = payload;
+    const { data: nextContacts } = payload;
 
     if (restart) {
-      this._rawContacts = {};
+      _rawContacts = {};
     }
 
-    for (const contact of contacts) {
-      this._rawContacts[contact.id] = contact;
+    for (const contact of nextContacts) {
+      _rawContacts[contact.id] = contact;
     }
-    this.setState({
-      contacts: Object.values(this._rawContacts),
-      hasPreviousPage,
-      hasNextPage,
-      refreshing: false,
-    });
-  }
-
-  onPressItem = async (id: string) => {
-    // tslint:disable-next-line no-console
-    console.log('onPress', id);
-    this.props.navigation.navigate('ContactDetail', { id });
-  }
+    setContacts(Object.values(_rawContacts));
+    setHasPreviousPage(payload.hasPreviousPage);
+    setHasNextPage(payload.hasNextPage);
+    setRefreshing(false);
```

Comment:
These aren't going to be batched IIRC (four renders)

Reply:
I got rid of one

![tenor](https://user-images.githubusercontent.com/9664363/84981034-7597b800-b0e8-11ea-9f7a-87d8441879e8.gif)